content/src/content/docs/docs/ai/getting-started.mdx

```diff
@@ -0,0 +1,466 @@
+# Install one (or more) provider integrations
+yarn add @effect/ai-openai
+
+# Also add the core Effect package (if not already installed)
+yarn add effect
+```
+
+</TabItem>
+
+<TabItem label="Bun" icon="bun">
+
+```sh showLineNumbers=false
+# Install the base package for the core abstractions (always required)
+bun add @effect/ai
+
+# Install one (or more) provider integrations
+bun add @effect/ai-openai
+
+# Also add the core Effect package (if not already installed)
+bun add effect
+```
+
+</TabItem>
+
+</Tabs>
+
+## Define an LLM Interaction
+
+First let's define a simple interaction with an LLM:
+
+```ts twoslash
+import { Completions } from "@effect/ai"
+import { Console, Effect } from "effect"
+
+// Using `Completions` will add it to your program's requirements
+//
+//          ┌─── Effect<void, AiError, Completions>
+//          ▼
+const generateDadJoke = Effect.gen(function*() {
+  // Extract the `Completions` service from the Effect environment
+  const completions = yield* Completions.Completions
+  // Use the `Completions` service to generate some text
+  return yield* completions.create("Generate a hilarious dad joke").pipe(
+    // Log the response text to the console
+    Effect.andThen((response) => Console.log(response.text))
+  )
+})
+```
+
+The above program uses the `Completions` service from the base `@effect/ai` package to describe using an LLM to create a text completion (in this case generating a hilarious dad joke). The result of the text generation is then logged to the console.
+
+<Aside type="note" title="Declarative LLM Interactions">
+  Notice that the above code does not know or care which LLM provider (OpenAi, Anthropic, etc.) will be used. Instead, we focus on _what_ we want to accomplish (i.e. our business logic), not _how_ to accomplish it.
+</Aside>
+
+## Select an LLM Model
+
+Next, we need to select which provider we want to use to satisfy the `Completions` requirement of our `generateDadJoke` program:
+
+```ts twoslash {1,17,25,27} collapse={5-10}
+import { OpenAiCompletions } from "@effect/ai-openai"
+import { Completions } from "@effect/ai"
+import { Console, Effect } from "effect"
+
+const generateDadJoke = Effect.gen(function*() {
+  const completions = yield* Completions.Completions
+  return yield* completions.create("Generate a hilarious dad joke").pipe(
+    Effect.andThen((response) => Console.log(response.text))
+  )
+})
+
+// Create an `AiModel` which provides a concrete implementation of 
+// `Completions` and requires an `OpenAiClient`
+//
+//      ┌─── AiModel<Completions, OpenAiClient>
+//      ▼
+const Gpt4o = OpenAiCompletions.model("gpt-4o")
+
+// Building an `AiModel` adds its requirements to the program
+//
+//     ┌─── Effect<void, AiError, OpenAiClient>
+//     ▼
+const main = Effect.gen(function*() {
+  // Build the `AiModel` into a `Provider`
+  const gpt4o = yield* Gpt4o
+  // Provide the implementation of `Completions` to `generateDadJoke`
+  yield* gpt4o.provide(generateDadJoke)
```

Comment:
```suggestion
  const res = yield* gpt4o.provide(generateDadJoke)
```

Reply:
Because `generateDadJoke` returns `void` in my sample program, I'm intentionally not binding these to a return variable.